src/utils/special-cases.ts

```diff
@@ -6,7 +6,7 @@ import { Job } from '../node-file-trace';
 import { Ast } from './types';
 type Node = Ast['body'][0]
 
-const specialCases: Record<string, (o: SpecialCaseOpts) => void> = {
+const specialCases: Record<string, (o: SpecialCaseOpts) => Promise<void> | void> = {
```

Comment:
@ijjk I think the [drop in codecov](https://user-images.githubusercontent.com/229881/179892820-2310570a-b6ae-4ca3-b4d7-d941dce2f5b9.png) indicates a bug.

This test is no longer hitting the typescript special case:

https://github.com/vercel/nft/blob/bbb2bb8cacf2b36da32cd6a6037f0ba80ea8b915/test/integration/typescript.js#L3

I think `require.resolve()` is suppose to be considered a dep but its now considered an asset with this PR.

Reply:
Hmm, actually not sure now that I see other tests using it to test assets

https://github.com/vercel/nft/blob/bbb2bb8cacf2b36da32cd6a6037f0ba80ea8b915/test/unit/require-resolve/input.js#L1

I'm worried there might be other code in the wild that depends on this behavior.

Reply:
I found the commit the typescript test was introduced: https://github.com/vercel/nft/commit/c596d9a6de9c9156553accba8383e27e354feffb

Perhaps @guybedford can provide some thoughts here 🙂

Reply:
I went ahead and updated the analysis to treat `require.resolve()` the same as `require()` (aka keep it a dependency instead of changing to an asset) since its common to pass the result to `child_process.spawn()` or something else that would need all dependencies.

See [189de9d](https://github.com/vercel/nft/pull/304/commits/189de9df031d31a45e60ba43c36ac13961e12e8c)